app/assets/javascripts/discourse/app/controllers/sidebar-section-form.js

```diff
@@ -229,27 +246,45 @@ export default Controller.extend(ModalFunctionality, {
       return new Section({
         title: this.model.title,
         publicSection: this.model.public,
+        sectionType: this.model.section_type,
         links: A(
-          this.model.links.map(
-            (link) =>
-              new SectionLink({
-                router: this.router,
-                icon: link.icon,
-                name: link.name,
-                value: link.value,
-                id: link.id,
-              })
-          )
+          this.model.links
+            .filter((link) => link.segment === "primary")
+            .map((link) => {
+              this.nextObjectId = this.nextObjectId + 1;
+              return this.initLink(link);
+            })
```

Comment:
```suggestion
          this.model.links
            .reduce((acc, link) => {
              if (link.segment === "primary") {
                this.nextObjectId++;
                acc.push(this.initLink);
              }

              return acc
            }, [])
```

Slightly more efficient way instead of looping through everything twice.